cdap-app-fabric/src/main/java/co/cask/cdap/security/TokenSecureStoreRenewer.java

```diff
@@ -70,16 +63,49 @@
   private Long updateInterval;
 
   @Inject
-  TokenSecureStoreUpdater(YarnConfiguration hConf, CConfiguration cConf,
+  TokenSecureStoreRenewer(YarnConfiguration hConf, CConfiguration cConf,
                           LocationFactory locationFactory,
                           co.cask.cdap.api.security.store.SecureStore secureStore) {
     this.hConf = hConf;
     this.locationFactory = locationFactory;
     this.secureStore = secureStore;
-    secureExplore = cConf.getBoolean(Constants.Explore.EXPLORE_ENABLED) && UserGroupInformation.isSecurityEnabled();
+    this.secureExplore = cConf.getBoolean(Constants.Explore.EXPLORE_ENABLED)
+      && UserGroupInformation.isSecurityEnabled();
+  }
+
+  /**
+   * Returns the minimum update interval for the delegation tokens.
+   * @return The update interval in milliseconds.
+   */
+  public long getUpdateInterval() {
+    if (updateInterval == null) {
+      // we want to lazily call this (as opposed to in the constructor), because sometimes we use an instance of
+      // TokenSecureStoreUpdater without scheduling updates. For instance, when launching a program or from
+      // ImpersonationHandler.
+      updateInterval = calculateUpdateInterval();
+    }
+    return updateInterval;
+  }
+
+  @Override
+  public void renew(String application, RunId runId, SecureStoreWriter secureStoreWriter) throws IOException {
+    Credentials credentials = createCredentials();
+    UserGroupInformation currentUser = null;
+    try {
+      currentUser = UserGroupInformation.getCurrentUser();
+    } catch (IOException e) {
+      // this shouldn't happen
+      LOG.debug("Cannot determine current user", e);
```

Comment:
It is intentional as it is not really going to happen. And if it does happen, there is nothing much can be done, so no differences in throwing exception or logging.